dv/management/commands/import_news.py

```diff
@@ -42,12 +41,24 @@ def _save(self, item):
             news.summary = item['summary']
             news.image = item['image'].replace('http://', 'https://')
             news.is_partnership = item['is_partnership'] == 'yes'
-            if item['project_id']:
-                news.project_id = item['project_id'].strip()
+            project_id = item.get("project_id")
+            if project_id:
+                project = Project.objects.filter(code=project_id).exists()
+                if project:
+                    news.project_id = project_id.strip()
+                else:
+                    self.stderr.write(f"Project code: {project_id} doesn't exist!")
             news.save()
 
-            for prg in item['programme_id'].split(', '):
-                news.programmes.add(prg.upper().strip())
+            item_programmes = item.get("programme_id")
```

Comment:
Python is dynamically typed, therefore you don't need to specify the variables' types.
Python is strongly typed, which means that variables always have a type.

Because of these two feature, we should be careful when choosing the name of a variable; the variable should suggest the data type it holds.

`item_programmes` is a string, then it becomes a list - not very easy to read.